tests/integration/test_healthchecks.py

```diff
@@ -97,3 +99,56 @@ def test_readiness_depends_on_aggregator_being_full(mini_sentry, relay):
     finally:
         # Authentication failures would fail the test
         mini_sentry.test_failures.clear()
+
+
+def test_readiness_disk_spool(mini_sentry, relay):
+    try:
+        temp = tempfile.mkdtemp()
+        dbfile = os.path.join(temp, "buffer.db")
+
+        project_key = 42
+        mini_sentry.add_full_project_config(project_key)
+        # Set the broken config, so we won't be able to dequeue the envelopes.
+        config = mini_sentry.project_configs[project_key]["config"]
+        ds = config.setdefault("dynamicSampling", {})
+        ds.setdefault("rules", [])
+        ds.setdefault("rulesV2", []).append(
+            {
+                "condition": {
+                    "op": "and",
+                    "inner": [
+                        {"op": "glob", "name": "releases", "value": ["1.1.1", "1.1.2"]}
+                    ],
+                },
+                "samplingValue": {"strategy": "sampleRate", "value": 0.7},
+                "type": "trace",
+                "id": 1,
+                "timeRange": {
+                    "start": "2022-10-10T00:00:00.000000Z",
+                    "end": "2022-10-20T00:00:00.000000Z",
+                },
+                "decayingFn": {"function": "linear", "decayedSampleRate": 0.9},
+            }
+        )
+
+        relay_config = {
+            "spool": {
+                "envelopes": {"path": dbfile, "max_memory_size": 0, "max_disk_size": 0}
+            },
+        }
+
+        relay = relay(
+            mini_sentry,
+            relay_config,
+            wait_health_check=True,
+        )
+
+        # These events will consume all the disk sapce and we will report not ready.
+        relay.send_event(project_key)
+        relay.send_event(project_key)
+
+        response = wait_get(relay, "/api/relay/healthcheck/ready/")
+        assert response.status_code == 503
+    finally:
+        # Authentication failures would fail the test
+        mini_sentry.test_failures.clear()
```

Comment:
@iker-barriocanal our testing harness has some magic which makes every test fail as soon as the relay instances logs an error to mini_sentry. So if you _expect_ error events, you have to explicitly clear this list.

@olksdr this would also work without a `try..finally` though, right?

Reply:
Yeah, I just re-used existing tests, to make them exactly same for uniformity.